build/changelog/ChangelogBuilder.php

```diff
@@ -0,0 +1,149 @@
+<?php
+namespace Aws\Build\Changelog;
+
+/**
+ * @internal
+ */
+
+class ChangelogBuilder 
+{
+  
+  public $newServiceFlag = False;
+
```

Comment:
I agree. The base directory should remain the same throughout a changelog build, so passing it as an argument to the class's constructor would be preferable.